service/dap/server.go

```diff
@@ -1150,12 +1150,35 @@ func (s *Server) doCommand(command string) {
 	stopped.Body.AllThreadsStopped = true
 
 	if err == nil {
-		stopped.Body.ThreadId = state.SelectedGoroutine.ID
+		if state.SelectedGoroutine != nil {
+			stopped.Body.ThreadId = state.SelectedGoroutine.ID
+		} else {
+			// If there is no selectedGoroutine, get the list of goroutines and select the first one.
```

Comment:
This isn't the correct behavior, if there is no selectedGoroutine that means that the current thread is not executing a goroutine. It would be better to set ThreadId = 0 in that case.

Reply:
Sounds good. The ThreadId for the stopped event is optional in the DAP spec, so it probably makes the most sense to not set the ThreadId in this case.

Reply:
Yeah this seems like a UI issue for the editors that are using the results of this call.